app/models/experimentation/reminder_template.rb

```diff
@@ -5,5 +5,13 @@ class ReminderTemplate < ActiveRecord::Base
 
     validates :message, presence: true
     validates :remind_on_in_days, presence: true, numericality: {only_integer: true}
+
+    validate :unique_message_per_group
+
+    def unique_message_per_group
+      if self.class.find_by(message: message, treatment_group: treatment_group)
+        errors.add(:message, "already exists in this treatment group")
+      end
```

Comment:
Can we use a uniqueness validation with a scope instead? https://guides.rubyonrails.org/active_record_validations.html#uniqueness